src/SDL12_compat.c

```diff
@@ -6596,9 +6711,14 @@ DECLSPEC void SDLCALL
 SDL_GL_SwapBuffers(void)
 {
     if (VideoWindow20) {
+        /* Some applications, e.g. Awesomenauts, play with glXMakeCurrent() behind out backs and break SwapBuffers() */
+        if (SDL12Compat_GetHintBoolean("SDL12COMPAT_FORCE_GL_SWAPBUFFER_CONTEXT", SDL_FALSE)) {
```

Comment:
How cheap will this be? Can't cache in some state or flag?

Reply:
It wasn't expensive enough that I noticed on my system, but I've cached it nevertheless, as it definitely could be more expensive on others.